service/controller/v25/resource/tccp/resource.go

```diff
@@ -126,6 +138,15 @@ func (r *Resource) searchMasterInstanceID(ctx context.Context, cr v1alpha1.AWSCo
 						aws.String(key.ClusterID(cr)),
 					},
 				},
+				{
+					Name: aws.String("instance-state-name"),
```

Comment:
This was there in the first place but I dropped it with the latest refactorings at some point. Turns out it is necessary. I bring it back but with the different instance states considered.